.github/workflows/test.yml

```diff
@@ -128,3 +141,19 @@ jobs:
 
       - name: Run check for type
         run: tox -e type --skip-pkg-install
+
+  # https://github.com/marketplace/actions/alls-green#why
+  check: # This job does nothing and is only used for the branch protection
```

Comment:
I have always been calling this "pass", not "check". That is a name I'd associate with a linting job, not for if all jobs passed.

Reply:
@henryiii yeah, naming of the ducks is hard. The linting association is weird, though. The statuses that show up at the bottom of PRs are coming (mostly) from the Checks API, for example — all of them are checks, not just linting ones.
I originally wanted to call this [`gate`](https://gating.dev) but a colleague convinced me to go for something more generic, which I now regret. So I've gone for using `check` in the examples, and now it's spread everywhere. And I keep using it because it's easier to update things across multiple projects if they look the same :man_shrugging:

Some people rename the job id, others add a custom `name` value (I saw `all-green` in PyCA, `required-checks-pass` by hynek, `success` by nedbat, I think somebody used `all-successful` and probably some others that I didn't care to remember). FWIW, I'll change to whatever @layday will want. Most folks just stick with `check` and don't care too much.

Reply:
I like "checks-pass" or "required-checks-pass" -- they get the point across. "check" I too would associate with linting and "pass" I'd have trouble deciphering.

Reply:
Let's go with "required-checks-pass" unless Henry disagrees.

Reply:
Done! Resolving.

Reply:
That works! I've been using "pass" in other places, and it works well in repos where there are multiple workflows - meson-python has three, for example, so there's things like tests-pass and docs-pass. tests-checks and docs-checks doesn't work nearly as well.

Reply:
Yes, many people don't immediately realize that this is per workflow. OTOH, with the current state of the reusable workflows, I'm leaning toward merging all the bits in a single one for having one gate for everything.

Reply:
> meson-python has three, for example, so there's things like tests-pass and docs-pass

So I've gone to check those workflows and saw that the aggregation jobs are now commented out and the previous implementation was faulty: [mesonbuild/meson-python@`1ef1721` (#221)](https://github.com/mesonbuild/meson-python/pull/221/commits/1ef1721772e505b869b7ef924a38556d8ed0b2bf#r1038060140).